bc_obps/registration/schema/v1/facility_designated_operation_timeline.py

```diff
@@ -0,0 +1,23 @@
+from registration.models.facility_designated_operation_timeline import FacilityDesignatedOperationTimeline
+from ninja import FilterSchema, ModelSchema, Field
+from typing import Optional
+
+
+class FacilityDesignatedOperationTimelineOut(ModelSchema):
+    facility__name: str = Field(..., alias="facility.name")
+    facility__type: str = Field(..., alias="facility.type")
+    facility__bcghg_id: Optional[str] = Field(None, alias="facility.bcghg_id")
+
+    class Meta:
+        model = FacilityDesignatedOperationTimeline
+        fields = ['id', 'status']
+
+
+class FacilityDesignatedOperationTimelineFilterSchema(FilterSchema):
+    # NOTE: we could simply use the `q` parameter to filter by name and ... but,
+    # due to this issue: https://github.com/vitalik/django-ninja/issues/1037 mypy is unhappy so I'm using the `json_schema_extra` parameter
+    # If we want to achieve more by using the `q` parameter, we should use it and ignore the mypy error
+    facility__bcghg_id: Optional[str] = Field(None, json_schema_extra={'q': 'facility__bcghg_id__icontains'})
+    facility__name: Optional[str] = Field(None, json_schema_extra={'q': 'facility__name__icontains'})
+    facility__type: Optional[str] = Field(None, json_schema_extra={'q': 'facility__type__icontains'})
+    status: Optional[str] = Field(None, json_schema_extra={'q': 'status__icontains'})
```

Comment:
I believe using double underscores (dunders) is very specific to Python. Since we are working with these value on the front end, I recommend using a single underscore. Alternatively, we can use camelCase like Giraffe does, but it's not consistent with the approach we've been using in Moose from the beginning.

Reply:
It looks like we need the dunders for the filter schema to work. If I change to a single underscore on the FE, then the URL params end up with a single underscore and filtering breaks. I could maybe change some of the params logic to add the underscore back in, but I think that might be more confusing. Thoughts?

Reply:
Filter schema only needs the dunder for `json_schema_extra` (dunder is the way Django reaches the related object), but I think we can change the name to whatever we want, like `whatever_name: Optional[str] = Field(None, json_schema_extra={'q': 'facility__name__icontains'}). 
If this is not the case, please ignore my comment :)

Reply:
That's what I had thought too, but it wasn't working :( Thanks for your help!